tools/corpus-refresh/refresh.py

```
@@ -0,0 +1,536 @@
+    # Client-side filter: only keep papers whose last revision is >= since.
+    since_aware = since if since.tzinfo is not None else since.replace(tzinfo=timezone.utc)
+    filtered: list[ArxivPaper] = []
+    for p in papers:
+        try:
+            updated_dt = datetime.fromisoformat(p.updated)
+            if updated_dt.tzinfo is None:
+                updated_dt = updated_dt.replace(tzinfo=timezone.utc)
+            if updated_dt >= since_aware:
+                filtered.append(p)
+        except (ValueError, AttributeError):
+            filtered.append(p)  # keep if we can't parse the date
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)